Задание для кандидатов. Выполнить ревью.sql

```diff
@@ -0,0 +1,123 @@
+create procedure syn.usp_ImportFileCustomerSeasonal
+	@ID_Record int
+AS
+set nocount on
+begin
+	declare @RowCount int = (select count(*) from syn.SA_CustomerSeasonal)
+	declare @ErrorMessage varchar(max)
+
+-- Проверка на корректность загрузки
+	if not exists (
+	select 1
+	from syn.ImportFile as f
+	where f.ID = @ID_Record
+		and f.FlagLoaded = cast(1 as bit)
+	)
+		begin
+			set @ErrorMessage = 'Ошибка при загрузке файла, проверьте корректность данных'
+
+			raiserror(@ErrorMessage, 3, 1)
+			return
+		end
+
+	--Чтение из слоя временных данных
+	select
+		c.ID as ID_dbo_Customer
+		,cst.ID as ID_CustomerSystemType
+		,s.ID as ID_Season
+		,cast(cs.DateBegin as date) as DateBegin
+		,cast(cs.DateEnd as date) as DateEnd
+		,c_dist.ID as ID_dbo_CustomerDistributor
+		,cast(isnull(cs.FlagActive, 0) as bit) as FlagActive
+	into #CustomerSeasonal
+	from syn.SA_CustomerSeasonal cs
+		join dbo.Customer as c on c.UID_DS = cs.UID_DS_Customer
+			and c.ID_mapping_DataSource = 1
+		join dbo.Season as s on s.Name = cs.Season
+		join dbo.Customer as c_dist on c_dist.UID_DS = cs.UID_DS_CustomerDistributor
+			and cd.ID_mapping_DataSource = 1
+		join syn.CustomerSystemType as cst on cs.CustomerSystemType = cst.Name

```

Comment:
Комментарий 6
Все виды join указываются явно
```suggestion
		inner join dbo.Customer as c on c.UID_DS = cs.UID_DS_Customer
			and c.ID_mapping_DataSource = 1
		inner join dbo.Season as s on s.Name = cs.Season
		inner join dbo.Customer as c_dist on c_dist.UID_DS = cs.UID_DS_CustomerDistributor
			and cd.ID_mapping_DataSource = 1
		inner join syn.CustomerSystemType as cst on cs.CustomerSystemType = cst.Name

```